Devx 142 feat/customized theme realm

roshanramangiri opened the pull request:
Please review this code and suggest me the necessary changes.

roshanramangiri committed:
DEV-142 Added Successfully running Keycloak Realm Import Docker and Modified the repo structure

roshanramangiri committed:
refactoring done, added new variables

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm secrets managed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm removing .env

susmita-karkii commented:
I didn't see deletion protection on alb, Is it not required?
Encyrption on RDS?

rajeshthapaliya commented:
Please look over the certificate section

saugat86 commented on .env:
```diff
@@ -0,0 +1,6 @@
+db_username               = "keycloak"
+db_password               = "secrectpassword"
```
Remove hardcoded db secrets and add on GitHub secrets and pull that secrets from GitHub secrets and also store this secrets on secrets manager.

saugat86 commented on main.tf:
```diff
@@ -1,151 +1,105 @@
-  ttl     = "86400"
+  name    = var.domain_name
+  ttl     = "300"
   zone_id = "Z03028901EPUSX1K65JBK"
```
Pass zone_id value using tfvars.

roshanramangiri commented:
@saugat86 dai, please review this.

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm wf added, structure changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm wf added, structure changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm permission added

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm oidc role changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm oidc role changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm oidc role changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm oidc role changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm oidc role changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm oidc role changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm oidc role changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm oidc role changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm oidc role changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm oidc role changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm oidc role changed

roshanramangiri committed:
DEVX-142-feat/customized_theme_realm oidc workflow changed

roshanramangiri commented:
@saugat86 I have changes as suggested by you.
i have,
- added workflow
- added github secrets
- utilizing .tfvars for values that keeps changing
- restructured the codes

Also check the documentation of this from the ticket

saugat86 commented on infra/modules/fargate/main.tf:
```diff
@@ -95,7 +101,7 @@ resource "aws_ecs_task_definition" "default" {
       options = {
         awslogs-group : "/ecs/keycloak"
-        awslogs-region : "us-east-1"
+        awslogs-region : "us-east-2"
```
Why did you use us-east-2?

saugat86 commented on infra/modules/alb/main.tf:
```diff
@@ -89,31 +85,14 @@ resource "aws_lb_listener" "https" {
+}
 
-# resource "aws_lb_listener" "https" {
-#   load_balancer_arn = aws_lb.alb.arn
```
Explain why this is commented?

saugat86 commented on .github/workflows/infra-deploy.yml:
```diff
@@ -0,0 +1,71 @@
+
+      - name: Terraform plan
+        run: |
+          cd ${{ env.working_dir }}
```
Post plan output while you are creating pull request. So that I can look more into this.

saugat86 commented on .github/workflows/infra-deploy.yml:
```diff
@@ -0,0 +1,71 @@
+    pull-requests: write
+
+env:
+  AWS_REGION: "us-east-2"
```
Please use us-east-1

saugat86 commented on scripts/push.sh:
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+REGION="us-east-2"
```
use us-east-1

saugat86 commented on infra/dev.tfvars:
```diff
@@ -0,0 +1,13 @@
+name                    = "keycloak"
+region                  = "us-east-2"
+owner                   = "Roshan Raman Giri"
```
remove specific flags and use generic other than that

saugat86 commented on infra/dev.tfvars:
```diff
@@ -0,0 +1,13 @@
+name                    = "keycloak"
+region                  = "us-east-2"
```
Use us-east-1 region.

saugat86 commented on infra/dev.tfvars:
```diff
@@ -0,0 +1,13 @@
+db_port                 = 5432
+db_multi_az             = false
+db_allocated_storage    = 20
+profile                 = "default"
```
Why did you use profile for this?

saugat86 commented on scripts/push.sh:
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+REGION="us-east-2"
+IMAGE_NAME="keycloak"
+MODULE_PATH=$MODULE_PATH
+aws ecr get-login-password --region $REGION | docker login --username AWS --password-stdin $AWS_ACCOUNT_ID.dkr.ecr.$REGION.amazonaws.com
+docker build -t $IMAGE_NAME:latest  $MODULE_PATH/../../../customization/
+docker tag $IMAGE_NAME:latest $AWS_ACCOUNT_ID.dkr.ecr.$REGION.amazonaws.com/$IMAGE_NAME:latest
+docker push $AWS_ACCOUNT_ID.dkr.ecr.$REGION.amazonaws.com/$IMAGE_NAME:latest
```
Add build phase on github action inorder to create docker image and upload it to ECR repository then only use terraform deploy pipeline.
